src/sentry/integrations/msteams/card_builder/installation.py

```diff
@@ -0,0 +1,94 @@
+from sentry.utils.http import absolute_uri
+
+from .base import (
+    ActionType,
+    ColumnWidth,
+    MSTeamsMessageBuilder,
+    TextSize,
+    TextWeight,
+    VerticalContentAlignment,
+)
+
+MSTEAMS_CONFIGURE_URL = "/extensions/msteams/configure/?signed_params={signed_params}"
+TEAM_INSTALLTION_TITLE = "Welcome to Sentry for Microsoft Teams"
+TEAM_INSTALLATION_DESCRIPTION = "You can use Sentry for Microsoft Teams to get notifications that allow you to assign, ignore, or resolve directly in your chat."
+TEAM_INSTALLATION_INSTRUCTION = (
+    "Please click **Complete Setup** to finish the setup process."
+    " Don't have a Sentry account? [Sign Up](https://sentry.io/signup/)."
+)
+TEAM_INSTALLATION_BUTTON = "Complete Setup"
+
+PERSONAL_INSTALLATION_TITLE = "Personal Installation of Sentry"
+PERSONAL_INSTALLATION_INSTRUCTION = (
+    "It looks like you have installed Sentry as a personal app."
+    " Sentry for Microsoft Teams needs to be added to a team. Please add"
+    ' Sentry again, and select "Add to a team" from the "Add" button\'s list arrow'
+)
+
+INSTALLATION_CONFIRMATION_TITLE = "Installation for {organization_name} is successful"
+INSTALLATION_CONFIRMATION_INSTRUCTION = (
+    "Now that setup is complete, you can continue by configuring alerts."
+)
+INSTALLATION_CONFIRMATION_BUTTON = "Add Alert Rules"
+ALERT_RULE_URL = "organizations/{organization_slug}/alerts/rules/"
+
+
+class MSTeamsInstallationTitleMessageBuilder(MSTeamsMessageBuilder):
+    def get_title_block(self, text: str) -> Any:
+        return self.get_column_set_block(
+            self.get_column_block(self.get_logo_block()),
+            self.get_column_block(
+                self.get_text_block(text, size=TextSize.LARGE, weight=TextWeight.BOLDER),
+                width=ColumnWidth.STRECH,
+                verticalContentAlignment=VerticalContentAlignment.CENTER,
+            ),
+        )
+
+
+class MSTeamsTeamInstallationMessageBuilder(MSTeamsInstallationTitleMessageBuilder):
+    def __init__(self, signed_params: str):
+        self.url = MSTEAMS_CONFIGURE_URL.format(signed_params=signed_params)
+
+    def build(self) -> Any:
+        return self._build(
+            title=self.get_title_block(TEAM_INSTALLTION_TITLE),
+            text=self.get_text_block(TEAM_INSTALLATION_DESCRIPTION),
+            fields=[self.get_text_block(TEAM_INSTALLATION_INSTRUCTION)],
+            actions=[
+                self.get_action_block(
+                    ActionType.OPEN_URL, title=TEAM_INSTALLATION_BUTTON, url=self.url
+                )
+            ],
+        )
+
+
+class MSTeamsPersonalIntallationMessageBuilder(MSTeamsInstallationTitleMessageBuilder):
+    def build(self) -> Any:
+        return self._build(
+            title=self.get_title_block(PERSONAL_INSTALLATION_TITLE),
+            text=self.get_text_block(PERSONAL_INSTALLATION_INSTRUCTION),
+        )
+
+
+class MSTeamsInstallationConfirmationMessageBuilder(MSTeamsInstallationTitleMessageBuilder):
```

Comment:
ex: 
```
def get_confirmation_message():
   return new MSTeamsMessageBuilder().build(title=..., actions=...)
```

Reply:
Summarizing the discussion we had - It does not make sense to have a class hierarchy and a class for each different type of message. Instead just have one message builder which takes care of the boilerplate card creation logic. Along with this use the static methods like `get_xxx_block` to create the cards inside the existing functions already present in `card_builder.py`.

Reply:
Remove all the specific message classes to be just functions. We only have one MSTeamsMessageBuilder class which builds the card now.